lib/std/os/linux.zig

```diff
@@ -1200,7 +1200,7 @@ pub fn ioctl(fd: fd_t, request: u32, arg: usize) usize {
     return syscall3(.ioctl, @bitCast(usize, @as(isize, fd)), request, arg);
 }
 
-pub fn signalfd4(fd: fd_t, mask: *const sigset_t, flags: i32) usize {
+pub fn signalfd(fd: fd_t, mask: *const sigset_t, flags: i32) usize {
```

Comment:
> There are two underlying Linux system calls: signalfd() and the more recent signalfd4().  The former system call does not implement  a  flags       argument.  The latter system call implements the flags values described above.  Starting with glibc 2.9, the signalfd() wrapper function will       use signalfd4() where it is available.

Reply:
I was confused over that situation on the original signalfd PR, but I don't understand what *should* we do on this case.

Reply:
It's a good question: what *is* the purpose of std/os/linux.zig? Should it match what the kernel does? or what glibc does?
At the moment I think its sort of trying to match glibc, so that it can be swapped out with libc-based routines.

@andrewrk might need your opinion here.

Reply:
It should match what the kernel does. Abstraction of the differences between the kernel and libcs  goes in the posix layer (currently `std.os`)